Remove npm cache clean logic in versioned test runner

[NOTE]: # ( ^^ Provide a general summary of the request in the title above. ^^ )

## Is your feature request related to a problem? Please describe.

We are getting failed versioned test runs and the actual error is getting masked on attempts to clean the npm cache. See this [example](https://github.com/newrelic/node-newrelic/runs/6613742956?check_suite_focus=true)

```sh
2022-05-26T17:12:33.4459591Z stderr
2022-05-26T17:12:33.4460756Z npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
2022-05-26T17:12:33.4461934Z npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
2022-05-26T17:12:33.4463164Z npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
2022-05-26T17:12:33.4464381Z npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
2022-05-26T17:12:33.4465628Z npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
2022-05-26T17:12:33.4467140Z npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
2022-05-26T17:12:33.4468559Z npm WARN deprecated uglify-es@3.3.9: support for ECMAScript is superseded by `uglify-js` as of v3.13.0
2022-05-26T17:12:33.4469199Z npm ERR! process terminated
2022-05-26T17:12:33.4469921Z npm ERR! signal SIGTERM
2022-05-26T17:12:33.4470256Z 
2022-05-26T17:12:33.4470908Z npm ERR! A complete log of this run can be found in:
2022-05-26T17:12:33.4471712Z npm ERR!     /home/runner/.npm/_logs/2022-05-26T17_11_19_769Z-debug-0.log
2022-05-26T17:12:33.4472565Z Clearing cache and retrying failed install...
2022-05-26T17:12:33.4473349Z npm ERR! As of npm@5, the npm cache self-heals from corruption issues
2022-05-26T17:12:33.4474304Z npm ERR!   by treating integrity mismatches as cache misses.  As a result,
2022-05-26T17:12:33.4475023Z npm ERR!   data extracted from the cache is guaranteed to be valid.  If you
2022-05-26T17:12:33.4475981Z npm ERR!   want to make sure everything is consistent, use `npm cache verify`
2022-05-26T17:12:33.4476698Z npm ERR!   instead.  Deleting the cache can only make npm go slower, and is
2022-05-26T17:12:33.4477637Z npm ERR!   not likely to correct any problems you may be encountering!
2022-05-26T17:12:33.4478149Z npm ERR! 
2022-05-26T17:12:33.4479135Z npm ERR!   On the other hand, if you're debugging an issue with the installer,
2022-05-26T17:12:33.4479887Z npm ERR!   or race conditions that depend on the timing of writing to an empty
2022-05-26T17:12:33.4481151Z npm ERR!   cache, you can use `npm install --cache /tmp/empty-cache` to use a
2022-05-26T17:12:33.4482433Z npm ERR!   temporary cache instead of nuking the actual one.
2022-05-26T17:12:33.4482753Z npm ERR! 
2022-05-26T17:12:33.4483334Z npm ERR!   If you're sure you want to delete the entire cache, rerun this command
2022-05-26T17:12:33.4483776Z npm ERR!   with --force.
2022-05-26T17:12:33.4483932Z 
2022-05-26T17:12:33.4484185Z npm ERR! A complete log of this run can be found in:
2022-05-26T17:12:33.4484788Z npm ERR!     /home/runner/.npm/_logs/2022-05-26T17_12_32_922Z-debug-0.log
2022-05-26T17:12:33.4485297Z Failed to clear cache: Error: Failed to execute npm cache clean
2022-05-26T17:12:33.4486166Z     at ChildProcess.exitHandler (/home/runner/work/node-newrelic/node-newrelic/node_modules/@newrelic/test-utilities/lib/versioned/runner.js:149:17)
2022-05-26T17:12:33.4486718Z     at ChildProcess.emit (node:events:527:28)
2022-05-26T17:12:33.4487200Z     at Process.ChildProcess._handle.onexit (node:internal/child_process:291:12)
2022-05-26T17:12:33.4487475Z 
2022-05-26T17:12:33.4487697Z ===============================================================
```

It is very hard to see why this attempt to install a package is failing.  Based on the message above they recommend to no longer clean the npm cache.

--- a/lib/versioned/runner.js
+++ b/lib/versioned/runner.js
@@ -14,7 +14,6 @@ const cp = require('child_process')
 // -) Exit codes less than zero mean this script failed.
 // 0) Exit codes equal to zero mean everything worked.
 
-const INSTALL_ATTEMPT_LIMIT = 3
 const CHILD_TIMEOUT = 60 * 1000 // 1 minute
 const CHILD_KILL_TIMEOUT = 10 * 1000 // 10 seconds
 
@@ -72,9 +71,9 @@ function installPackages(cb) {
     return cb()
   }
 
-  tryInstall(0)
+  tryInstall()
 
-  function tryInstall(attempts) {
+  function tryInstall() {
     let args = [
       'install',
       '--no-save', // do not update package file
@@ -85,26 +84,9 @@ function installPackages(cb) {
     args = args.concat(packages)
     spawn('npm', args, function installHandler(err) {
       if (err) {
-        if (++attempts < INSTALL_ATTEMPT_LIMIT) {
-          console.error('Clearing cache and retrying failed install...')
-          setTimeout(function installRetry() {
-            spawn('npm', ['cache', 'clean'], function cacheHandler(e) {
-              if (e) {
-                const error = new Error('Failed to clear cache: ' + e.stack)
-                error.code = -Math.abs(e.code || 0xbeef)
-                return cb(error)
-              }
-              tryInstall(attempts)
-            })
-          }, 5000 * attempts) // 5 seconds * number of attempts
-          return
-        }
-
-        const error = new Error('Failed to install packages: ' + err.stack)
-        error.code = -Math.abs(err.code)
-        return cb(error)
+        err.code = -Math.abs(err.code)
       }
-      cb()
+      cb(err)
     })
   }
 }

